subprojects/core/src/main/java/org/gradle/execution/commandline/CommandLineTaskConfigurer.java

```diff
@@ -94,4 +87,10 @@ private List<String> configureTasksNow(Collection<Task> tasks, List<String> argu
         }
         return remainingArguments;
     }
+
+    private void allowOneOfEach(CommandLineParser parser, List<OptionDescriptor> oppositeOptions) {
+        for (int i = 0; i < oppositeOptions.size(); i = i + 2) {
+            parser.allowOneOf(oppositeOptions.get(i).getName(), oppositeOptions.get(i+1).getName());
```

Comment:
🎨 It would be more readable to use a `List<Pair>` here instead of fiddling with the iteration index on a structure created in another object.